Sources/join.swift

```diff
@@ -17,6 +17,12 @@ import Dispatch
  - Returns: A new promise that resolves once all the provided promises resolve.
 */
 public func join<T>(promises: Promise<T>...) -> Promise<[T]> {
+    return join(promises)
+}
+
+public func join<T>(promises: [Promise<T>]) -> Promise<[T]> {
+    guard !promises.isEmpty else { return Promise<[T]>([]) }
```

Comment:
I'm uncertain what the contract here is - should an empty array of promises to `join` immediately resolve or should an "invalid use" error be produced like below?